test/e2e/allocator_test.go

```diff
@@ -70,9 +73,23 @@ func TestAllocator(t *testing.T) {
 	if !assert.Nil(t, err) {
 		return
 	}
-	response, err := client.Post(requestURL, "application/json", bytes.NewBuffer(body))
-	if !assert.Nil(t, err) {
-		return
+
+	// wait for the allocation system to come online
+	var response *http.Response
```

Comment:
@pooneh-m for interest's sake - this was what (seems to have) solved the flakiness issue. My theory is that there was a race condition wherein the backing pods for the service would come up after this test was run, but only sometimes (probably depending on e2e test order).

Reply:
Thanks for the fix. Makes sense.